lib/plugins/aws/provider/awsProvider.js

```diff
@@ -111,8 +123,9 @@ const impl = {
   addEnvironmentProfile: (results, prefix) => {
     if (prefix) {
       const profile = process.env[`${prefix}_PROFILE`];
-      impl.addProfileCredentials(results, profile);
+      return impl.addProfileCredentials(results, profile);
     }
+    return BbPromise.resolve();
```

Comment:
Same here, it'll be more migration friendly to just wrap it with `BbPromise.try`